### PR TITLE
Update to draft 32

### PR DESCRIPTION
**Pull request by Ralith**
Incorporates all documented changes between draft 29 and the wg master, excluding h3. Nothing major.

### CODE REVIEW

File: interop/src/main.rs [hunk 1]
```diff
@@ -65,9 +65,9 @@ impl From<&Alpn> for Vec<Vec<u8>> {
     fn from(alpn: &Alpn) -> Vec<Vec<u8>> {
         match alpn {
             Alpn::H3 => vec![quinn_h3::ALPN.into()],
-            Alpn::Hq => vec![b"hq-29"[..].into()],
+            Alpn::Hq => vec![b"hq-31"[..].into()],
```
Comment (djc) [inline interop/src/main.rs:68]: I wonder, should we actually leave this at 29 for a while? From the mailing list it looks like large deployments are sticking to the draft-29 version identifiers even as they deploy improvements from 30 and later.

File: README.md [hunk 2]
```diff
@@ -45,7 +45,7 @@ Quinn was created and is maintained by Dirkjan Ochtman and Benjamin Saunders.
 
 ## Status
 
-- [x] QUIC draft 27 with TLS 1.3
+- [x] QUIC draft 31 with TLS 1.3
```
Comment (djc) [inline README.md:48]: Another commit mentions draft 32, but you used 31 first. Which is it? Are they different (enough)?

**Review by djc — COMMENTED** [review]
Thanks for doing this, LGTM.

File: quinn-proto/src/lib.rs [hunk 3]
```diff
@@ -117,7 +117,7 @@ pub mod fuzzing {
     }
 }
 /// The QUIC protocol version implemented
-const VERSION: u32 = 0xff00_001d;
+const VERSION: u32 = 0xff00_001f;
```
Comment (Matthias247) [inline quinn-proto/src/lib.rs:120]: Please leave it at the old version - maybe with a comment.
Browsers like Safari and Chrome will now starting to use Chrome, but they will all announce Draft 29 version support even if they are 30+.
With this change things would be needlessly imcompatible.

File: README.md (hunk 2 above)
Comment (Ralith) [reply README.md:48]: 2bff792 is from after draft 31 was cut, but draft 32 has not been formally released yet.

Comment (djc) [reply README.md:48]: I have email from the mailing list announcing draft 32, from Oct 21st. How do you determine it hasn't been released?

**Comment by Ralith** [comment]
Removed the global version number bump, and relaxed the compatible version check to support a range rather than a single version.

File: README.md (hunk 2 above)
Comment (Ralith) [reply README.md]: Whoops, good catch.

File: interop/src/main.rs (hunk 1 above)
Comment (Ralith) [reply interop/src/main.rs]: Done.

**Review by djc — APPROVED** [review]

**Pull request merged**